### PR TITLE
Model linespan endpoint

### PR DESCRIPTION
**Pull request by Free-Quarks**
This is for an added endpoint that takes in a zip file containing a python script and returns the line span the dynamics should be under. 

It currently uses an Openai LLM for this, but we will replace that with a custom local model once it's ready.

### CODE REVIEW

**Review by myedibleenso — CHANGES_REQUESTED** [review]
@Free-Quarks , have you tested this locally with the other project deps?  We're using pydantic v2 in this project.  I've found that to be incompatible with langchain's prompt templates.  To ensure skema isn't going to hit the same issue, can you please add a test that instantiates a prompt template?

**Comment by myedibleenso** [comment]
Downgrading pydantic to a different major version might cause unexpected behavior.  If you run into this issue in the test, maybe you can get around using the langchain prompt template by creating your own prompt string.

**Comment by Free-Quarks** [comment]
> @Free-Quarks , have you tested this locally with the other project deps? We're using pydantic v2 in this project. I've found that to be incompatible with langchain's prompt templates. To ensure skema isn't going to hit the same issue, can you please add a test that instantiates a prompt template?

Thanks for pointing this out! I am using the generic output parser from langchain, not their pydantic one, so hopefully it won't be an issue. I'm using pydantic just to simplify the final serialized json for FastAPI. But I will test it today.

**Comment by Free-Quarks** [comment]
@myedibleenso So I was already running on pydantic 2.3, but I also updated to 2.4.1 and tested locally and it still works on that as well. So there shouldn't be an issue with pydantic v2!

File: skema/rest/tests/test_llms.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    human_message_prompt = HumanMessagePromptTemplate.from_template(human_template)
+
+    # combining the templates for a chat template
+    chat_prompt = ChatPromptTemplate.from_messages([system_message_prompt, human_message_prompt])
+
+    # formatting the prompt with input variables
+    formatted_prompt = chat_prompt.format_prompt(code=code, format_instructions = format_instructions).to_messages()
+
+    # running the model
+    output = openai(formatted_prompt)
```
Comment (myedibleenso) [inline skema/rest/tests/test_llms.py:53]: I don't think we'll want this API key to be available during CI (lots of calls, risk of leakage, etc.).  Can we either a) mock this part or b) test only the prompt's format/structure?

**Review by myedibleenso — CHANGES_REQUESTED** [review]

File: skema/rest/tests/test_llms.py (hunk 1 above)
Comment (Free-Quarks) [reply skema/rest/tests/test_llms.py:53]: Understandable, I'll look into that!

Comment (Free-Quarks) [reply skema/rest/tests/test_llms.py]: Okay! I am testing the prompts and mocked up the LLM response to test the parser!

**Review by myedibleenso — APPROVED** [review]
LGTM!  Thanks, @Free-Quarks , for adding those unit tests.

**Pull request merged**